qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java

```diff
@@ -0,0 +1,284 @@
+        manageLocationTagsPage.enterLocationTagDescription("editedlocationTagDescription");
+    }
+
+    @And("the user clicks on the retire location tag button")
+    public void userClicksOnTheRetireLocationTagButton() {
+        manageLocationTagsPage.retireLocation();
+    }
+
+    @Then("the system retires the location tag")
+    public void systemRetiresTheLocationTag() {
+        assertTrue(textExists("Location Tag Retired"));
+    }
+
+    @And("the user clicks the restore location tag icon")
+    public void userClicksTheRestoreLocationTagIcon() {
+        manageLocationTagsPage.restoreLocation();
+        assertTrue(textExists("Location Tag restored"));
+    }
+
+    @And("the user clicks on the delete location tag forever icon")
+    public void userClicksOnTheDeleteLocationTagForeverIcon() {
+        manageLocationTagsPage.deleteLocationTag();
+    }
+
+    @Then("the system deletes the location tag forever")
+    public void systemDeletesTheLocationTagForever() {
+        assertTrue(textExists("Location Tag Deleted"));
+    }
+
+    @When("a user clicks on manage location link")
+    public void userClicksOnManageLocation() {
+        manageLocationsPage = configureMetadataPage.goToManageLocations();
+    }
+
+    @Then("the system loads the manage location page")
+    public void systemLoadsTheManageLocationPage() {
+        assertTrue(textExists("Manage Locations"));
+    }
+
+    @And("the user clicks on add new location button")
+    public void userClicksOnAddNewLocationButton() {
+        manageLocationsPage.goToAddNewLocation();
+        assertTrue(textExists("Add New Location"));
+    }
+
+    @And("the user fills the form")
+    public void userFillsAddNewLocationForm() {
+        manageLocationsPage.enterLocationName(LOCATION_NAME);
+        manageLocationsPage.enterLocationDescription(LOCATION_DESCRIPTION);
+        manageLocationsPage.enterAddress1("Kampala");
+        manageLocationsPage.enterAddress2("Wakiso");
+        manageLocationsPage.enterCity("Kampala city");
+        manageLocationsPage.enterState("Central");
+        manageLocationsPage.enterCountry("Uganda");
+        manageLocationsPage.enterPostalCode("postalCode");
+        manageLocationsPage.selectLocationTag();
+    }
+
+    @Then("the user saves the location form")
+    public void userSavesAddNewLocationForm() {
+        manageLocationsPage.saveLocation();
+        // assertTrue(textExists("Location Saved"));
+    }
+
+    @And("the user clicks on the edit location icon")
+    public void userClicksOnTheEditLocationIcon() {
+        manageLocationsPage.goToEditLocationForm();
+    }
+
+    @And("the user fills in the prefered details in the edit location form")
+    public void userFillsInThePreferedDetailsInTheEditLocationForm() {
+
+    }
+
+    @And("the user clicks on the retire location button")
+    public void userClicksOnTheRetireLocationIcon() {
+        manageLocationsPage.retireLocation(LOCATION_NAME);
```

Comment:
Add a waiting element here before line 260 to listen or wait for the elements to be visible.